'GetOrCreateCache' operation takes too long when backedup by JDBC store

### Describe the bug

#15148 increased the socket timeouts for server DB tests to 10_000ms (default 2_000ms) to stabilize the tests. 

Nevertheless the `getOrCreateCache` against empty database with empty cache shouldn't take that long.The increased timeouts suggests there's a regression that needs to be investigated.

```
[ERROR] org.infinispan.server.persistence.ManagedConnectionOperations.testPutGetRemove(Database)[1] -- Time elapsed: 2.467 s <<< ERROR!
org.infinispan.client.hotrod.exceptions.TransportException:: java.net.SocketTimeoutException: org.infinispan.client.hotrod.impl.transport.netty.HeaderDecoder@7b7a6c41 timed out after 2000 ms
	at org.infinispan.client.hotrod.impl.Util.rewrap(Util.java:110)
	at org.infinispan.client.hotrod.impl.Util.await(Util.java:62)
	at org.infinispan.client.hotrod.impl.Util.await(Util.java:50)
	at org.infinispan.client.hotrod.impl.RemoteCacheManagerAdminImpl.getOrCreateCache(RemoteCacheManagerAdminImpl.java:95)
	at org.infinispan.server.test.api.HotRodTestClientDriver.create(HotRodTestClientDriver.java:138)
	at org.infinispan.server.test.api.HotRodTestClientDriver.create(HotRodTestClientDriver.java:119)
	at org.infinispan.server.persistence.ManagedConnectionOperations.testPutGetRemove(ManagedConnectionOperations.java:69)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at java.base/java.util.Optional.ifPresent(Optional.java:178)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.net.SocketTimeoutException: org.infinispan.client.hotrod.impl.transport.netty.HeaderDecoder@7b7a6c41 timed out after 2000 ms
	at org.infinispan.client.hotrod.impl.transport.netty.HeaderDecoder.checkForTimeouts(HeaderDecoder.java:116)
	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:156)
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472)
	at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:408)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:998)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

### Expected behavior

Test'll pass even with default timeout of 2_000s as they did with Infinsipan < 16.0.

### Actual behavior

The getOrCreateCache operation timeouts with the default timeout.

### How to Reproduce?

Comment out increased timeout property `infinispan.client.hotrod.socket_timeout` in the `server/tests/src/test/resources/database/mysql.properties` file and run the test:

```
mvn clean install -DskipTests=true
mvn clean verify -pl server/tests -Dit.test=ManagedConnectionOperations -Dorg.infinispan.test.database.types=mysql
```

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Infinispan version or git rev

main

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.6 (Red Hat 3.9.6-7) Maven home: /usr/share/maven Java version: 17.0.15, vendor: Red Hat, Inc., runtime: /usr/lib/jvm/java-17-openjdk Default locale: en_US, platform encoding: UTF-8 OS name: "linux", version: "6.15.4-100.fc41.x86_64", arch: "amd64", family: "unix"

### Additional information

_No response_

--- a/server/rollingupgradetests/src/test/java/org/infinispan/server/rollingupgrade/PersistenceRollingUpgradeIT.java
+++ b/server/rollingupgradetests/src/test/java/org/infinispan/server/rollingupgrade/PersistenceRollingUpgradeIT.java
@@ -22,6 +22,11 @@ public class PersistenceRollingUpgradeIT extends InfinispanSuite {
 
    @RegisterExtension
    public static RollingUpgradeHandlerExtension SERVERS =
-         RollingUpgradeHandlerExtension.from(PersistenceRollingUpgradeIT.class, PersistenceIT.EXTENSION_BUILDER,
-               RollingUpgradeTestUtil.getFromVersion(), RollingUpgradeTestUtil.getToVersion());
+         new RollingUpgradeHandlerExtension(
+               RollingUpgradeHandlerExtension.convertBuilder(
+                     PersistenceRollingUpgradeIT.class,
+                     PersistenceIT.EXTENSION_BUILDER,
+                     RollingUpgradeTestUtil.getFromVersion(),
+                     RollingUpgradeTestUtil.getToVersion()
+               ).nodeCount(2));
 }

--- a/server/tests/src/test/java/org/infinispan/server/persistence/PersistenceIT.java
+++ b/server/tests/src/test/java/org/infinispan/server/persistence/PersistenceIT.java
@@ -57,7 +57,7 @@ public class PersistenceIT extends InfinispanSuite {
 
    public static final InfinispanServerExtensionBuilder EXTENSION_BUILDER =
          InfinispanServerExtensionBuilder.config(System.getProperty(PersistenceIT.class.getName(), "configuration/PersistenceTest.xml"))
-               .numServers(2)
+               .numServers(1)
                .runMode(ServerRunMode.CONTAINER)
                .mavenArtifacts(getJdbcDrivers())
                .artifacts(getJavaArchive())

